src/Cli/dotnet/Commands/Test/VSTestForwardingApp.cs

```diff
@@ -38,14 +38,17 @@ private static string GetVSTestExePath()
         return Path.Combine(AppContext.BaseDirectory, VstestAppName);
     }
 
-    internal static (bool hasRootVariable, string rootVariableName, string rootValue) GetRootVariable()
+    internal static Dictionary<string, string> GetVSTestRootVariables()
     {
-        string rootVariableName = Environment.Is64BitProcess ? "DOTNET_ROOT" : "DOTNET_ROOT(x86)";
-        bool hasRootVariable = Environment.GetEnvironmentVariable(rootVariableName) != null;
-        string rootValue = hasRootVariable ? null : Path.GetDirectoryName(new Muxer().MuxerPath);
-
-        // We rename env variable to support --arch switch that relies on DOTNET_ROOT/DOTNET_ROOT(x86)
-        // We provide VSTEST_WINAPPHOST_ only in case of testhost*.exe removing VSTEST_WINAPPHOST_ prefix and passing as env vars.
-        return (hasRootVariable, $"VSTEST_WINAPPHOST_{rootVariableName}", rootValue);
```

Comment:
I'm double checking, this is fine because this logic isn't going to be used with old vstest.console.exe because it's guaranteed that we use vstest.console.exe that's part of SDK, and we will only merge this PR after inserting the fix on vstest side. Is that correct?

Reply:
Yes, the old way of doing this can be removed. It also does not seem that people use that env variable on its own.